Framework/Algorithms/CMakeLists.txt

```diff
@@ -830,7 +833,8 @@ enable_precompiled_headers ( inc/MantidAlgorithms/PrecompiledHeader.h SRC_FILES
 # Add the target for this directory
 add_library ( Algorithms ${SRC_FILES} ${INC_FILES})
 # Set the name of the generated library
-set_target_properties ( Algorithms PROPERTIES OUTPUT_NAME MantidAlgorithms )
+set_target_properties ( Algorithms PROPERTIES OUTPUT_NAME MantidAlgorithms
+						           COMPILE_DEFINITIONS "IN_MANTID_ALGORITHMS" )
```

Comment:
Interesting. I wasn't aware of this `"IN_MANTID_ALGORITHMS"`+ `MANTID_ALGORITHMS_DLL`.